File: 1-two-sum/1-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int n = nums.length;
        int []res = new int[2];
        Map<Integer,Integer> mp = new HashMap<>();
        for(int i = 0;i<n;i++){
            if(mp.containsKey(target - nums[i])){
                res[1] = i;
                res[0] = mp.get(target-nums[i]);
                return res;
            }
            else{
                mp.put(nums[i],i);
            }
        }
        return res;
    }
}

File: 1-two-sum/1-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        //Using HashSet: Time: O(n) and Space:O(n)
        vector<int> res;
        unordered_map<int,int>mp;
        for(int i =0;i<nums.size();i++)
        {
            if(mp.find(target-nums[i]) != mp.end()){
                res.push_back(i);
                res.push_back(mp[target-nums[i]]);
            }
            mp[nums[i]] = i;
        }
        return res;
    }
};